Makefile

```diff
@@ -90,8 +107,52 @@ qemu-gdb: GOFLAGS := $(GOFLAGS:-s=)
 qemu-gdb: $(APP)
 	$(QEMU) -kernel $(APP) -S -s
 
+qemu-img: $(APP).img
+	$(QEMU-img)
+
+qemu-img-gdb: $(APP).img
+	$(QEMU-img) -S -s
+
+#### AMD64 targets ####
+
+ifeq ($(TARGET),microvm)
+img: $(APP).img
+
+$(APP).bin: $(APP)
+	objcopy -j .text -j .rodata -j .shstrtab -j .typelink \
+	    -j .itablink -j .gopclntab -j .go.buildinfo -j .noptrdata -j .data \
+	    -j .bss --set-section-flags .bss=alloc,load,contents \
+	    -j .noptrbss --set-section-flags .noptrbss=alloc,load,contents \
+	    $(APP) -O binary $(APP).bin
+
+tools/mbr.bin: tools/mbr.s $(APP) $(APP).bin
+	OFFSET=200 && \
+	  SIZE=$$(stat -c %s $(APP).bin) && \
+	  ENTRY=0x$$(dd if=$(APP) bs=1 count=4 skip=24 | xxd -e -g4 | xxd -r | xxd -p) && \
```

Comment:
This is too complex/verbose for Makefile, it should be moved to a script in tools.